substrate/utils/frame/omni-bencher/src/command.rs

```diff
@@ -134,14 +134,6 @@ impl V1SubCommand {
 		match self {
 			V1SubCommand::Benchmark(V1BenchmarkCommand { sub }) => match sub {
 				BenchmarkCmd::Pallet(pallet) => {
-					if let Some(spec) = pallet.shared_params.chain {
-						return Err(format!(
-							"Chain specs are not supported. Please remove `--chain={spec}` and use \
-				`--runtime=<PATH>` instead"
-						)
-						.into());
-					}
-
```

Comment:
not sure if this is a good thing to remove,
because `pallet.shared_params.chain` is not passed bellow to `pallet.run_with_spec::<BlakeTwo256, HostFunctions>(None)`,

maybe for correctness we should pass it here?
```
pallet.run_with_spec::<BlakeTwo256, HostFunctions>(pallet.shared_params.chain)
```

in case of invalid args, there is another check for CLI args:
```
fn check_args(
		&self,
		chain_spec: &Option<Box<dyn ChainSpec>>,
	) -> std::result::Result<(), (ErrorKind, String)> {
		if self.runtime.is_some() && self.shared_params.chain.is_some() {
```

Reply:
My understanding is that given we call `check_args` inside the `pallet.run_with_spec` method, which handles already very well various cases of using `--chain` and `--runtime` together, or passsing a `Some(<Box<dyn ChainSpec>>)` to the method, we don't have to make this check anymore.